src/OpenTelemetry.Sampler.AWS/AWSXRayRemoteSampler.cs

```diff
@@ -43,20 +50,41 @@ internal AWSXRayRemoteSampler(TimeSpan pollingInterval, string endpoint)
 
     internal Timer RulePollerTimer { get; }
 
+    private RulesCache RulesCache { get; }
+
+    private Trace.Sampler FallbackSampler { get; }
+
     /// <summary>
     /// Initializes a <see cref="AWSXRayRemoteSamplerBuilder"/> for the sampler.
     /// </summary>
+    /// <param name="resource">an instance of <see cref="Resources.Resource"/>
+    /// to identify the service attributes for sampling. This resource should
+    /// be the same as what the OpenTelemetry SDK is configured with.</param>
     /// <returns>an instance of <see cref="AWSXRayRemoteSamplerBuilder"/>.</returns>
-    public static AWSXRayRemoteSamplerBuilder Builder()
+    public static AWSXRayRemoteSamplerBuilder Builder(Resource resource)
```

Comment:
I am not sure if Resource is the best way how to configure this values.
Based on what I see, you need Service name and cloud platform.
Can it be passed here?

Reply:
My aim of using a `Resource` here is the ease of use. Typically, an OTel user would configure the service name through either environment variable or a Resource. Also, we have a bunch of AWS resource detectors which can automatically detect the cloud platform. So I think passing in a resource here is better than explicitly passing service name and cloud platform values. 
Also, this way the sampler behavior will be consistent across other OTel SDK. For example OTel Java SDK: https://github.com/open-telemetry/opentelemetry-java-contrib/blob/main/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/AwsXrayRemoteSampler.java#L67

Reply:
@utpilla, FYI